### PR TITLE
chore: upgrade ariakit 3.0.4

### PR DESCRIPTION
**Pull request by mleralec**

### CODE REVIEW

File: docs/pages/components/modal.mdx [hunk 1]
```diff
@@ -78,7 +78,7 @@ function() {
       <Modal.Trigger as={Button} store={modal} onClick={() => setSize('auto')}>
         auto
       </Modal.Trigger>
-      <Modal store={modal} size={size} ariaLabel="size example">
+      <Modal store={modal} size={size} ariaLabel="size example" unmountOnHide>
```
Comment (theo-mesnil) [inline docs/pages/components/modal.mdx:81]: it's a test ? :D

File: packages/Modal/src/index.tsx [hunk 2]
```diff
@@ -12,18 +12,18 @@ import { Content } from './Content'
...
   size?: Size
-  store: Ariakit.DialogStore
 }
+
 export type ModalProps = CreateWuiProps<'div', ModalOptions>
 export type UseModal = Ariakit.DialogStore
 export type UseModalProps = Ariakit.DialogStoreProps & {
   /**
    * Call a function before closing the modal
+   * @deprecated use Modal.onClose instead
```
Comment (theo-mesnil) [inline packages/Modal/src/index.tsx:26]:
```suggestion
   * @deprecated use onClose on <Modal /> instead
```

**Review by theo-mesnil — APPROVED** [review]
🔥

File: docs/pages/components/modal.mdx (hunk 1 above)
Comment (mleralec) [reply docs/pages/components/modal.mdx:81]: Nop 😈

**Pull request merged**